src/Database/LSMTree/Internal/Run/Index/Compact/Construction.hs

```diff
@@ -3,21 +3,32 @@
 {-# LANGUAGE RecordWildCards     #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
+
+{- |
+  Incremental construction of a compact index yields chunks incrementally. These
+  chunks can be converted to a 'CompactIndex' once incremental construction is
+  finalised.
+  TODO
+
+  Incremental construction is an 'ST' computation that can be started using
+  'new', returning an 'MCompactIndex' structure that accumulates internal state.
+  'append'ing new pages to the 'MCompactIndex' /might/ yield 'Chunk's.
+  Incremental construction can be finalised with 'unsafeEnd', which yields both
+  a 'Chunk' (possibly) and 'FinalChunk'. Use 'fromChunks' on the 'Chunk's and
+  the 'FinalChunk' to construct a 'CompactIndex'.
+-}
 module Database.LSMTree.Internal.Run.Index.Compact.Construction (
-    -- $incremental
+    -- * Construction
     -- $construction
-    PageNo (..)
-  , MCompactIndex
+    MCompactIndex
+  , PageNo (..)
```

Comment:
Tiny thing: I know there are some differing opinions on whether re-exports are a good thing. Do you have a preference?

Reply:
If types like `PageNo` had a logical and stable place to import them from, I'd be against re-exporting them. But currently they are kind of just defined in some module where they are used, sometimes moved somewhere else.

Do you have a good place in mind? I don't really want an `Internal.Types` module, but maybe that's least bad?

Reply:
It's not something we have to change now, but it's a useful discussion in any case. Let's leave it here for now until the modules become more stable